tensorflow_datasets/structured/adult.py

```diff
@@ -0,0 +1,216 @@
+                    convert_to_string)),
+
+    ("final_weight", (tf.int32, convert_to_int)),
+    ("education", (tfds.features.ClassLabel(names=_EDUCATION_LABELS),
+                   convert_to_string)),
+
+    ("education_num", (tf.int32, convert_to_int)),
+    ("marital_status", (tfds.features.ClassLabel(names=_MARITAL_LABELS),
+                        convert_to_string)),
+
+    ("occupation", (tfds.features.ClassLabel(names=_OCCUPATION_LABELS),
+                    convert_to_string)),
+
+    ("relationship", (tfds.features.ClassLabel(names=_RELATION_LABELS),
+                      convert_to_string)),
+
+    ("race", (tfds.features.ClassLabel(names=_RACE_LABELS),
+              convert_to_string)),
+
+    ("sex", (tfds.features.ClassLabel(names=["Male", "Female"]),
+             convert_to_string)),
+
+    ("capital_gain", (tf.int32, convert_to_int)),
+    ("capital_loss", (tf.int32, convert_to_int)),
+    ("hours_per_week", (tf.int32, convert_to_int)),
+    ("native_country", (tfds.features.ClassLabel(names=_COUNTRY_LABELS),
+                        convert_to_string))
+])
+
+_URL_TRAIN = "https://archive.ics.uci.edu/ml/machine-learning-databases/adult/adult.data"
+_URL_TEST = "https://archive.ics.uci.edu/ml/machine-learning-databases/adult/adult.test"
+
+class Adult(tfds.core.GeneratorBasedBuilder):
+  """UCI Adult Dataset"""
+
+  VERSION = tfds.core.Version(
+      "0.0.1")
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            "prediction": tfds.features.ClassLabel(names=["<=50K", ">50K"]),
+            "features": {
+                name: dtype for name, (dtype, func) in FEATURE_DICT.items()
+            }
+        }),
+
+        supervised_keys=("features", "prediction"),
+        homepage='https://archive.ics.uci.edu/ml/datasets/Adult',
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    files = dl_manager.download({
+        "train_data": _URL_TRAIN,
+        "test_data": _URL_TEST
+    })
+
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            gen_kwargs={"file_path": files["train_data"]},
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TEST,
+            gen_kwargs={"file_path": files["test_data"]}
+        )
+    ]
+
+  def _generate_examples(self, file_path):
+    """Generate features and target given the directory path.
+    Args:
+      file_path: path where the csv file is stored
+    Yields:
+      The features and the target
+    """
+    with tf.io.gfile.GFile(file_path) as f:
+      raw_data = csv.reader(f)
+      for i, row in enumerate(raw_data):
+        if len(row) == 15:
+          prediction_val = row[-1].split('.')[0].strip()
+          rec = dict(zip(COLUMNS, map(lambda x: x.strip(), row[:-1])))
```

Comment:
Please remove this by list comprehension.

Reply:
I can't seem to find a way to use list comprehension here. What exactly do you want me to remove?

Reply:
There was some `pylint` issues. Then please disable that error message for this line if there is no other way to do this.

Reply:
I just ran `pylint` on the latest version of the PR and it is rated 10.0/10.0